packages/yourloops/components/alarm/alarms-content-configuration.hook.tsx

```diff
@@ -0,0 +1,234 @@
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import { UNITS_TYPE } from '../../lib/units/utils'
+import { useTeam } from '../../lib/team'
+import { Monitoring } from '../../models/monitoring'
+import React, { useState, useMemo, Dispatch } from 'react'
+import PatientUtils from '../../lib/patient/utils'
+import { Patient } from '../../lib/data/patient'
+
+export interface AlarmsContentConfigurationHookProps {
+  monitoring?: Monitoring
+  saveInProgress?: boolean
+  patient?: Patient
+  onClose?: () => void
+  onSave: (monitoring: Monitoring) => void
+}
+
+interface AlarmsContentConfigurationHookReturn {
+  isError: (value: number, lowValue: number, highValue: number) => boolean
+  lowBg: ValueErrorPair
+  setLowBg: React.Dispatch<ValueErrorPair>
+  veryLowBg: ValueErrorPair
+  setVeryLowBg: React.Dispatch<ValueErrorPair>
+  highBg: ValueErrorPair
+  setHighBg: React.Dispatch<ValueErrorPair>
+  nonDataTxThreshold: ValueErrorPair
+  setNonDataTxThreshold: React.Dispatch<ValueErrorPair>
+  outOfRangeThreshold: ValueErrorPair
+  setOutOfRangeThreshold: React.Dispatch<ValueErrorPair>
+  hypoThreshold: ValueErrorPair
+  setHypoThreshold: React.Dispatch<ValueErrorPair>
+  saveButtonDisabled: boolean
+  onBasicDropdownSelect: (value: string, setValue: React.Dispatch<{ value?: number, error: boolean }>) => void
+  save: () => void
+  resetToTeamDefaultValues: () => void
+  onChange: (value: number, lowValue: number, highValue: number, setValue: React.Dispatch<ValueErrorPair>) => void
+  MIN_HIGH_BG: number
+  MAX_HIGH_BG: number
+  MIN_VERY_LOW_BG: number
+  MAX_VERY_LOW_BG: number
+  MIN_LOW_BG: number
+  MAX_LOW_BG: number
+  PERCENTAGES: string[]
+  bgUnit: UNITS_TYPE
+}
+
+interface ValueErrorPair {
+  value?: number
+  error: boolean
+}
+
+export const MIN_HIGH_BG = 140
+export const MAX_HIGH_BG = 250
+export const MIN_VERY_LOW_BG = 40
+export const MAX_VERY_LOW_BG = 90
+export const MIN_LOW_BG = 50
+export const MAX_LOW_BG = 100
+export const PERCENTAGES = [...new Array(21)]
+  .map((_each, index) => `${index * 5}%`).slice(1, 21)
+
+const useAlarmsContentConfiguration = ({ monitoring, saveInProgress, onSave, patient }: AlarmsContentConfigurationHookProps): AlarmsContentConfigurationHookReturn => {
+  const bgUnit = monitoring?.parameters?.bgUnit ?? UNITS_TYPE.MGDL
+  const isInvalidPercentage = (value: number): boolean => {
+    return !PERCENTAGES.includes(`${value}%`)
+  }
+  const isError = (value: number, lowValue: number, highValue: number): boolean => {
```

Comment:
This function does not depend on any state or hook and can exported from an util file (could be named alarm-content-configuration.utils.tsx`)